### PR TITLE
Add FindChecked to MapById.

### PR DESCRIPTION
**Pull request by cschuet**

### CODE REVIEW

File: cartographer/mapping/id.h [hunk 1]
```diff
@@ -158,6 +158,16 @@ class MapById {
       }
     }
 
+    explicit ConstIterator(const MapById& map_by_id, const NodeId& id)
```
Comment (wohe) [inline cartographer/mapping/id.h]: `NodeId` -> `IdType`
Also: Add a test using a `SubmapId`?

File: cartographer/mapping/id.h [hunk 2]
```diff
@@ -225,6 +235,11 @@ class MapById {
     return IdType{trajectory_id, index};
   }
 
+  // Returs an iterator to the element at id or throws an exception.
```
Comment (wohe) [inline cartographer/mapping/id.h]: typo: `Returs` -> `Returns`
`id` -> `'id'`
We use a `CHECK`, so: `or throws an exception` -> `which must exist`.

File: cartographer/mapping/id.h (hunk 1 above)
Comment (cschuet) [reply cartographer/mapping/id.h]: Done

File: cartographer/mapping/id.h (hunk 2 above)
Comment (cschuet) [reply cartographer/mapping/id.h]: Done

**Pull request merged**